shared-libs/transitions/src/transitions/mark_for_outbound.js

```diff
@@ -48,8 +48,17 @@ const markForOutbound = (change) => {
       .then(sent => {
         if (sent) {
           // Successfully sent, outbound.send wrote to the infodoc
-
-          return db.sentinel.put(change.info);
+          return db.sentinel.get(change.info._id)
```

Comment:
I think this conflict resolution is a bit verbose to be here, and it doesn't treat an  eventual conflict directly - in your error, the status that the PUT request got was 409, and here you're only resolving for 404s (not found). 

I think the best solution here is to use the `infodoc` shared lib, which already has infodoc conflct resolution built in _and_ already has a `saveCompletedTasks` function  (https://github.com/medic/cht-core/blob/master/shared-libs/infodoc/src/infodoc.js#L285). This function doesn't receive parameters now, but that can be changed. 

Please let me know what you think!